File: lab/python/textos/maiusculas_e_minusculas.py
def minuscula_e_maiuscula1(texto):
    print("VERSÃO 1: upper e lower")
    print(texto.lower())
    print(texto.upper())


def minuscula_e_maiuscula2(texto):
    print("VERSÃO 2: ASCII manual")
    minusculas = ""
    maiusculas = ""

    for c in texto:
        if 'A' <= c <= 'Z':  # Se for maiúscula
            minusculas += chr(ord(c) + 32)
            maiusculas += c
        elif 'a' <= c <= 'z':  # Se for minúscula
            minusculas += c
            maiusculas += chr(ord(c) - 32)
        else:
            # Mantém outros símbolos (espaço, pontuação, etc.)
            minusculas += c
            maiusculas += c

    print(minusculas)
    print(maiusculas)


def minuscula_e_maiuscula3(texto):
    print("VERSÃO 3: maketrans (Usa mapeamento de substituição)")
    letras_maiusculas = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    letras_minusculas = "abcdefghijklmnopqrstuvwxyz"

    mapa_minusculas = str.maketrans(letras_maiusculas, letras_minusculas)
    mapa_maiusculas = str.maketrans(letras_minusculas, letras_maiusculas)

    print(texto.translate(mapa_minusculas))
    print(texto.translate(mapa_maiusculas))


testes = "Ola", "TESTE", "salve", "Python"
for strings in testes:
    minuscula_e_maiuscula1(strings)
    minuscula_e_maiuscula2(strings)
    minuscula_e_maiuscula3(strings)
    print()

File: lab/c/textos/maiusculas_e_minusculas.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

void minuscula_e_maiuscula1(const char *texto) {
    printf("VERSÃO 1: tolower e toupper\n");
    for (int i = 0; texto[i] != '\0'; i++)
        putchar(tolower((unsigned char)texto[i]));
    printf("\n");
    for (int i = 0; texto[i] != '\0'; i++)
        putchar(toupper((unsigned char)texto[i]));
    printf("\n");
}

void minuscula_e_maiuscula2(const char *texto) {
    printf("VERSÃO 2: ASCII manual\n");
    char minusculas[256] = "", maiusculas[256] = "";
    int i = 0;

    while (texto[i] != '\0') {
        char c = texto[i];
        if (c >= 'A' && c <= 'Z') {
            minusculas[i] = c + 32;
            maiusculas[i] = c;
        } else if (c >= 'a' && c <= 'z') {
            minusculas[i] = c;
            maiusculas[i] = c - 32;
        } else {
            minusculas[i] = c;
            maiusculas[i] = c;
        }
        i++;
    }
    minusculas[i] = '\0';
    maiusculas[i] = '\0';

    printf("%s\n", minusculas);
    printf("%s\n", maiusculas);
}

void minuscula_e_maiuscula3(const char *texto) {
    printf("VERSÃO 3: tabela de tradução\n");
    char minusculas[256] = "", maiusculas[256] = "";
    char letras_maiusculas[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    char letras_minusculas[] = "abcdefghijklmnopqrstuvwxyz";

    for (int i = 0; texto[i] != '\0'; i++) {
        char c = texto[i];
        int found = 0;
        for (int j = 0; j < 26; j++) {
            if (c == letras_maiusculas[j]) {
                minusculas[i] = letras_minusculas[j];
                maiusculas[i] = c;
                found = 1;
                break;
            } else if (c == letras_minusculas[j]) {
                minusculas[i] = c;
                maiusculas[i] = letras_maiusculas[j];
                found = 1;
                break;
            }
        }
        if (!found) {
            minusculas[i] = c;
            maiusculas[i] = c;
        }
    }
    printf("%s\n", minusculas);
    printf("%s\n", maiusculas);
}

int main(void) {
    const char *testes[] = {"Ola", "TESTE", "salve", "Python"};
    int n = sizeof(testes) / sizeof(testes[0]);

    for (int i = 0; i < n; i++) {
        minuscula_e_maiuscula1(testes[i]);
        minuscula_e_maiuscula2(testes[i]);
        minuscula_e_maiuscula3(testes[i]);
        printf("\n");
    }

    return 0;
}
